source/Engine.cpp

```diff
@@ -1692,9 +1732,42 @@ void Engine::CalculateStep()
 
 
 
+// Move a ship. Can be called concurrently.
+void Engine::MoveShip(const shared_ptr<Ship> &ship, vector<mutex> &bufferMutexes, vector<list<Visual>> &visualsBuffer,
+		vector<list<shared_ptr<Flotsam>>> &flotsamBuffer,
+		vector<list<shared_ptr<Ship>>> &shipBuffer, vector<vector<Projectile>> &projectileBuffer)
+{
+	// Find a free lock, and call MoveShip with the locked lists.
+	size_t id = std::hash<thread::id>{}(this_thread::get_id());
+	const int index = id % bufferMutexes.size();
+	if(bufferMutexes[index].try_lock())
+	{
+		MoveShip(ship, visualsBuffer[index], flotsamBuffer[index], shipBuffer[index], projectileBuffer[index]);
+		bufferMutexes[index].unlock();
+		return;
+	}
+	else
+	{
+		for(unsigned newIndex = 0; newIndex < bufferMutexes.size(); newIndex++)
+			if(bufferMutexes[newIndex].try_lock())
+			{
+				MoveShip(ship, visualsBuffer[newIndex], flotsamBuffer[newIndex], shipBuffer[newIndex], projectileBuffer[newIndex]);
+				bufferMutexes[newIndex].unlock();
+				return;
+			}
+	}
+	// If there is no free lock, fall back to the original guess.
+	bufferMutexes[index].lock();
+	MoveShip(ship, visualsBuffer[index], flotsamBuffer[index], shipBuffer[index], projectileBuffer[index]);
+	bufferMutexes[index].unlock();
+}
+
+
+
 // Move a ship. Also determine if the ship should generate hyperspace sounds or
 // boarding events, fire weapons, and launch fighters.
-void Engine::MoveShip(const shared_ptr<Ship> &ship)
+void Engine::MoveShip(const shared_ptr<Ship> &ship, list<Visual> &newVisuals, list<shared_ptr<Flotsam>> &newFlotsam,
+		list<shared_ptr<Ship>> &newShips, vector<Projectile> &newProjectiles)
 {
 	// Various actions a ship could have taken last frame may have impacted the accuracy of cached values.
 	// Therefore, determine with any information needs recalculated and cache it.
```

Comment:
Not quite in the scope of this PR maybe, but that comment is rather hard to read, so maybe it should be adjusted?